---
title: "DESeq Analysis on EGME small RNA data (New)"
output: html_document
---
Perform DESeq2 and related analysis on 6 different datasets:
* miRNA counts generated by sRNAde
* miRNA counts parsed from sRNAbench (the same data as above, but with more miRNAs)
* ncRNA counts parsed from sRNAbench
* tRNA counts parsed from sRNAbench
* cDNA counts parsed from sRNAbench
* a dataset contains all types of small RNAs using another dataset

For each of the dataset, perform the following preprocessing procedure:
1. remove Library_24 and Library_33, as by QC they have problems like low counts 
2. filter out small RNAs with less than 10 counts across all libraries

```{r setup, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
library(DESeq2)
library(pheatmap)
library(tidyverse)
library(readxl)
library(RColorBrewer)
library(EnhancedVolcano)

## set paths to read count files
# from sRNAde results, 80 miRNAs; used a resaved version for convenience
path_miRNA_sRNAde <- "mature_sense_re-saved.csv" 
# parsed from sRNAbench results
path_miRNA <- "mature_sense_counts.csv" # same as path_miRNA_sRNAde but with more miRNAs
path_ncRNA <- "Rnor_6_0_ncRNA_sense_counts.csv"
path_tRNA <- "Rnor_6_0_genomic_tRNA_sense_counts.csv"
path_cDNA <- "Rnor_6_0_cDNA_sense_counts.csv"
path_RNAcentral <- "Rnor_6_0_RNAcentral_sense_counts.csv" # another database


## build colData
groupPath <- "100.21 RNAseq summary 2020-07-06 DS.xlsx"
groups <- read_excel(groupPath)
dosages <- groups$`treatment (mg/kg)`
chips <- paste(groups$chip) # convert to string
folders <- paste(groups$folder) # convert to string
dates <- paste(groups$date) # convert to string
samples <- groups$`sample number`# libraries
colData <- data.frame(dosage=dosages, chip=chips, folder=folders, date=dates, sample=samples)
colData <- colData[colData$sample!=24,] # Remove library 24
colData <- colData[colData$sample!=33,] # Remove library 33
colData$dosage <- factor(colData$dosage, levels = c("0","50","60","75"))
```



### miRNA from sRNAde {.tabset}
We first analyze the miRNA read counts generated by sRNAde. 

```{r load_miRNA_sRNAde, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
readCounts <- read.csv(path_miRNA_sRNAde)
genes <- readCounts[, 1]
readCounts <- data.matrix(readCounts[, -1])
libraries <- colnames(readCounts)

readCounts <- matrix(as.numeric(readCounts), ncol=length(libraries))
rownames(readCounts) <- genes
colnames(readCounts) <- libraries
readCounts <- readCounts[, order(colnames(readCounts))]
readCounts <- readCounts[, !colnames(readCounts) %in% c("Library_24")] # Remove library 24
readCounts <- readCounts[, !colnames(readCounts) %in% c("Library_33")] # Remove library 33

rownames(colData) <- colnames(readCounts)

dds <- DESeqDataSetFromMatrix(countData=readCounts, colData=colData, design=~dosage)

# pre-filtering
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]

vsd <- varianceStabilizingTransformation(dds, blind = FALSE)
```

#### sample-wise heatmap
* rows and columns clustered
* label = library+dosages

```{r sample-wise_heatmap_miRNA_sRNAde, echo=FALSE}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste("Library",vsd$sample,"  -  Dosage",vsd$dosage)
colnames(sampleDistMatrix) <- NULL #vsd$dosage
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors, cluster_rows = T, cluster_cols = T)
```

#### PCA plots
* From top to bottom: (1) groups = dosage; (2) groups = chip; (3) groups = date; (4) groups = folder.
* The results seem to suggest the batch effect of chip (see chip 6) / date (see 2018-04-18) /folder.
* It is interesting that the datapoints labeled with dosage roughly form a spectrum on PC1.

```{r PCA_miRNA_sRNAde, echo=FALSE}
plotPCA(vsd, intgroup = c("dosage"))
plotPCA(vsd, intgroup = c("chip"))
plotPCA(vsd, intgroup = c("date"))
plotPCA(vsd, intgroup = c("folder"))
```

#### DESeq with Wald test
```{r DESeq_w_miRNA_sRNAde, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_wald <- DESeq(dds)
resultsNames(dds_wald)
```
* dosage 50mg vs 0mg results (only print the top 10 miRNAs ordered by smallest p-value first, same for below)
```{r DESeq_wr1_miRNA_sRNAde, echo=FALSE}
res_wald_50 <- results(dds_wald, name="dosage_50_vs_0")
res_wald_50 <- res_wald_50[order(res_wald_50$pvalue),]
res_wald_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_wr2_miRNA_sRNAde, echo=FALSE}
res_wald_60 <- results(dds_wald, name="dosage_60_vs_0")
res_wald_60 <- res_wald_60[order(res_wald_60$pvalue),]
res_wald_60[1:10,]
```
* dosage 75mg vs 0mg results 
```{r DESeq_wr3_miRNA_sRNAde, echo=FALSE}
res_wald_75 <- results(dds_wald, name="dosage_75_vs_0")
res_wald_75 <- res_wald_75[order(res_wald_75$pvalue),]
res_wald_75[1:10,]
```

#### Volcano plots for DESeq with Wald test

```{r volcano_wald_miRNA_sRNAde, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_wald_50,
    lab = rownames(res_wald_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_60,
    lab = rownames(res_wald_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_75,
    lab = rownames(res_wald_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```

#### DESeq with likelihood ratio test
Since we are running the log likelihood test, the p-values should be the same for all dosage groups. Log2FoldChange will be, however, different for each dosage group.
```{r DESeq_l_miRNA_sRNAde, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_lrt <- DESeq(dds, test="LRT", reduced=~1)
resultsNames(dds_lrt)
```
* dosage 50mg vs 0mg results (only print the top 10 miRNAs ordered by smallest p-value first, same for below)
```{r DESeq_lr1_miRNA_sRNAde, echo=FALSE}
res_lrt_50 <- results(dds_lrt, name="dosage_50_vs_0")
res_lrt_50 <- res_lrt_50[order(res_lrt_50$pvalue),]
res_lrt_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_lr2_miRNA_sRNAde, echo=FALSE}
res_lrt_60 <- results(dds_lrt, name="dosage_60_vs_0")
res_lrt_60 <- res_lrt_60[order(res_lrt_60$pvalue),]
res_lrt_60[1:10,]
```
* dosage 75mg vs 0mg results 
```{r DESeq_lr3_miRNA_sRNAde, echo=FALSE}
res_lrt_75 <- results(dds_lrt, name="dosage_75_vs_0")
res_lrt_75 <- res_lrt_75[order(res_lrt_75$pvalue),]
res_lrt_75[1:10,]
```

#### Volcano plots for DESeq with likelihood ratio test
```{r volcano_lrt_miRNA_sRNAde, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_lrt_50,
    lab = rownames(res_lrt_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_60,
    lab = rownames(res_lrt_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_75,
    lab = rownames(res_lrt_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```

#### heatmap of counts
* columns clustered 
* show miRNAs of top 20 mean counts
```{r DESeq_hm_miRNA_sRNAde, echo=FALSE}
select_wald <- order(rowMeans(counts(dds_wald, normalized=TRUE)),
                decreasing=TRUE)[1:20] # top 20
df_wald <- as.data.frame(colData(dds_wald)[,"dosage"])
colnames(df_wald) <- "dosage"
rownames(df_wald) <- colnames(dds_wald)
pheatmap(assay(vsd)[select_wald,], cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df_wald) # looks strange
```




***
***



### miRNA counts parsed from sRNAbench results {.tabset}
Repeat the above analysis on miRNA counts parsed from sRNAbench results
```{r load_miRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
readCounts <- read.csv(path_miRNA)
genes <- readCounts[, 1]
readCounts <- data.matrix(readCounts[, -1])
libraries <- colnames(readCounts)

readCounts <- matrix(as.numeric(readCounts), ncol=length(libraries))
rownames(readCounts) <- genes
colnames(readCounts) <- libraries
readCounts <- readCounts[, order(colnames(readCounts))]
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_24")] # Remove library 24
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_33")] # Remove library 33

rownames(colData) <- colnames(readCounts)

dds <- DESeqDataSetFromMatrix(countData=readCounts, colData=colData, design=~dosage)

# pre-filtering
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]

vsd <- varianceStabilizingTransformation(dds, blind = FALSE)
```

#### sample-wise heatmap
* rows and columns clustered
* label = library+dosages

```{r sample-wise_heatmap_miRNA, echo=FALSE}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste("Library",vsd$sample,"  -  Dosage",vsd$dosage)
colnames(sampleDistMatrix) <- NULL #vsd$dosage
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors, cluster_rows = T, cluster_cols = T)
```

#### PCA plots
* From top to bottom: (1) groups = dosage; (2) groups = chip; (3) groups = date; (4) groups = folder.
* The results seem to suggest the batch effect of chip (see chip 6) / date (see 2018-04-18) /folder.
* Compared to the 80-miRNA results from sRNAde, the plots look much different.
* The spectrum-like distribution of data over dosage PC1 also appears here.
```{r PCA_miRNA, echo=FALSE}
plotPCA(vsd, intgroup = c("dosage"))
plotPCA(vsd, intgroup = c("chip"))
plotPCA(vsd, intgroup = c("date"))
plotPCA(vsd, intgroup = c("folder"))
```

#### DESeq with Wald test
Some miRNAs have NA padj values (not shown). NA padj values are generated per mechanism "If a row is filtered by automatic independent filtering, for having a low mean normalized count, then only the adjusted p value will be set to NA."
```{r DESeq_w_miRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_wald <- DESeq(dds)
resultsNames(dds_wald)
```
* dosage 50mg vs 0mg results (only print the top 10 miRNAs ordered by smallest p-value first, same for below)
```{r DESeq_wr1_miRNA, echo=FALSE}
res_wald_50 <- results(dds_wald, name="dosage_50_vs_0")
res_wald_50 <- res_wald_50[order(res_wald_50$pvalue),]
res_wald_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_wr2_miRNA, echo=FALSE}
res_wald_60 <- results(dds_wald, name="dosage_60_vs_0")
res_wald_60 <- res_wald_60[order(res_wald_60$pvalue),]
res_wald_60[1:10,]
```
* dosage 75mg vs 0mg results
```{r DESeq_wr3_miRNA, echo=FALSE}
res_wald_75 <- results(dds_wald, name="dosage_75_vs_0")
res_wald_75 <- res_wald_75[order(res_wald_75$pvalue),]
res_wald_75[1:10,]
```

#### Volcano plots for DESeq with Wald test
```{r volcano_wald_miRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_wald_50,
    lab = rownames(res_wald_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_60,
    lab = rownames(res_wald_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_75,
    lab = rownames(res_wald_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```



#### DESeq with likelihood ratio test
Since we are running the log likelihood test, the p-values should be the same for all dosage groups. Log2FoldChange will be, however, different for each dosage group.
```{r DESeq_l_miRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_lrt <- DESeq(dds, test="LRT", reduced=~1)
resultsNames(dds_lrt)
```
* dosage 50mg vs 0mg results (only print the top 10 miRNAs ordered by smallest p-value first, same for below)
```{r DESeq_lr1_miRNA, echo=FALSE}
res_lrt_50 <- results(dds_lrt, name="dosage_50_vs_0")
res_lrt_50 <- res_lrt_50[order(res_lrt_50$pvalue),]
res_lrt_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_lr2_miRNA, echo=FALSE}
res_lrt_60 <- results(dds_lrt, name="dosage_60_vs_0")
res_lrt_60 <- res_lrt_60[order(res_lrt_60$pvalue),]
res_lrt_60[1:10,]
```
* dosage 75mg vs 0mg results 
```{r DESeq_lr3_miRNA, echo=FALSE}
res_lrt_75 <- results(dds_lrt, name="dosage_75_vs_0")
res_lrt_75 <- res_lrt_75[order(res_lrt_75$pvalue),]
res_lrt_75[1:10,]
```


#### Volcano plots for DESeq with likelihood ratio test
```{r volcano_lrt_miRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_lrt_50,
    lab = rownames(res_lrt_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_60,
    lab = rownames(res_lrt_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_75,
    lab = rownames(res_lrt_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```

#### heatmap of counts
* columns clustered 
* show miRNAs of top 20 mean counts
```{r DESeq_hm_miRNA, echo=FALSE}
select_wald <- order(rowMeans(counts(dds_wald, normalized=TRUE)),
                decreasing=TRUE)[1:20] # top 20
df_wald <- as.data.frame(colData(dds_wald)[,"dosage"])
colnames(df_wald) <- "dosage"
rownames(df_wald) <- colnames(dds_wald)
pheatmap(assay(vsd)[select_wald,], cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df_wald) # looks strange
```




***
***



### tRNA counts parsed from sRNAbench results {.tabset}
Repeat the above analysis on tRNA counts parsed from sRNAbench results
```{r load_tRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
readCounts <- read.csv(path_tRNA)
genes <- readCounts[, 1]
readCounts <- data.matrix(readCounts[, -1])
libraries <- colnames(readCounts)

readCounts <- matrix(as.numeric(readCounts), ncol=length(libraries))
rownames(readCounts) <- genes
colnames(readCounts) <- libraries
readCounts <- readCounts[, order(colnames(readCounts))]
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_24")] # Remove library 24
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_33")] # Remove library 33

rownames(colData) <- colnames(readCounts)

dds <- DESeqDataSetFromMatrix(countData=readCounts, colData=colData, design=~dosage)

# pre-filtering
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]

vsd <- varianceStabilizingTransformation(dds, blind = FALSE)
```

#### sample-wise heatmap
* rows and columns clustered
* label = library+dosages

```{r sample-wise_heatmap_tRNA, echo=FALSE}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste("Library",vsd$sample,"  -  Dosage",vsd$dosage)
colnames(sampleDistMatrix) <- NULL #vsd$dosage
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors, cluster_rows = T, cluster_cols = T)
```

#### PCA plots
* From top to bottom: (1) groups = dosage; (2) groups = chip; (3) groups = date; (4) groups = folder
* The results seem to suggest the batch effect of chip (see chip 6) / date (see 2018-04-18).
* The spectrum-like distribution of data over dosage PC1 still observable.
* Library_06 as an outlier?
```{r PCA_tRNA, echo=FALSE}
plotPCA(vsd, intgroup = c("dosage"))
plotPCA(vsd, intgroup = c("chip"))
plotPCA(vsd, intgroup = c("date"))
plotPCA(vsd, intgroup = c("folder"))
```

#### DESeq with Wald test
Some miRNAs have NA padj values (not shown). NA padj values are generated per mechanism "If a row is filtered by automatic independent filtering, for having a low mean normalized count, then only the adjusted p value will be set to NA."
```{r DESeq_w_tRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_wald <- DESeq(dds)
resultsNames(dds_wald)
```
* dosage 50mg vs 0mg results (only print the top 10 tRNAs ordered by smallest p-value first, same for below)
```{r DESeq_wr1_tRNA, echo=FALSE}
res_wald_50 <- results(dds_wald, name="dosage_50_vs_0")
res_wald_50 <- res_wald_50[order(res_wald_50$pvalue),]
res_wald_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_wr2_tRNA, echo=FALSE}
res_wald_60 <- results(dds_wald, name="dosage_60_vs_0")
res_wald_60 <- res_wald_60[order(res_wald_60$pvalue),]
res_wald_60[1:10,]
```
* dosage 75mg vs 0mg results
```{r DESeq_wr3_tRNA, echo=FALSE}
res_wald_75 <- results(dds_wald, name="dosage_75_vs_0")
res_wald_75 <- res_wald_75[order(res_wald_75$pvalue),]
res_wald_75[1:10,]
```

#### Volcano plots for DESeq with Wald test
```{r volcano_wald_tRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_wald_50,
    lab = rownames(res_wald_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_60,
    lab = rownames(res_wald_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_75,
    lab = rownames(res_wald_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```


#### DESeq with likelihood ratio test
Since we are running the log likelihood test, the p-values should be the same for all dosage groups. Log2FoldChange will be, however, different for each dosage group.
```{r DESeq_l_tRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_lrt <- DESeq(dds, test="LRT", reduced=~1)
resultsNames(dds_lrt)
```
* dosage 50mg vs 0mg results (only print the top 10 tRNAs ordered by smallest p-value first, same for below)
```{r DESeq_lr1_tRNA, echo=FALSE}
res_lrt_50 <- results(dds_lrt, name="dosage_50_vs_0")
res_lrt_50 <- res_lrt_50[order(res_lrt_50$pvalue),]
res_lrt_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_lr2_tRNA, echo=FALSE}
res_lrt_60 <- results(dds_lrt, name="dosage_60_vs_0")
res_lrt_60 <- res_lrt_60[order(res_lrt_60$pvalue),]
res_lrt_60[1:10,]
```
* dosage 75mg vs 0mg results 
```{r DESeq_lr3_tRNA, echo=FALSE}
res_lrt_75 <- results(dds_lrt, name="dosage_75_vs_0")
res_lrt_75 <- res_lrt_75[order(res_lrt_75$pvalue),]
res_lrt_75[1:10,]
```


#### Volcano plots for DESeq with likelihood ratio test
```{r volcano_lrt_tRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_lrt_50,
    lab = rownames(res_lrt_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_60,
    lab = rownames(res_lrt_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_75,
    lab = rownames(res_lrt_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```


#### heatmap of counts
* columns clustered 
* show tRNAs of top 20 mean counts
```{r DESeq_hm_tRNA, echo=FALSE}
select_wald <- order(rowMeans(counts(dds_wald, normalized=TRUE)),
                decreasing=TRUE)[1:20] # top 20
df_wald <- as.data.frame(colData(dds_wald)[,"dosage"])
colnames(df_wald) <- "dosage"
rownames(df_wald) <- colnames(dds_wald)
pheatmap(assay(vsd)[select_wald,], cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df_wald) # looks strange
```



***
***



### cDNA counts parsed from sRNAbench results {.tabset}
Repeat the above analysis on cDNA counts parsed from sRNAbench results
```{r load_cDNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
readCounts <- read.csv(path_cDNA)
genes <- readCounts[, 1]
readCounts <- data.matrix(readCounts[, -1])
libraries <- colnames(readCounts)

readCounts <- matrix(as.numeric(readCounts), ncol=length(libraries))
rownames(readCounts) <- genes
colnames(readCounts) <- libraries
readCounts <- readCounts[, order(colnames(readCounts))]
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_24")] # Remove library 24
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_33")] # Remove library 33

rownames(colData) <- colnames(readCounts)

dds <- DESeqDataSetFromMatrix(countData=readCounts, colData=colData, design=~dosage)

# pre-filtering
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]

vsd <- varianceStabilizingTransformation(dds, blind = FALSE)
```

#### sample-wise heatmap
* rows and columns clustered
* label = library+dosages

```{r sample-wise_heatmap_cDNA, echo=FALSE}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste("Library",vsd$sample,"  -  Dosage",vsd$dosage)
colnames(sampleDistMatrix) <- NULL #vsd$dosage
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors, cluster_rows = T, cluster_cols = T)
```

#### PCA plots
* From top to bottom: (1) groups = dosage; (2) groups = chip; (3) groups = date; (4) groups = folder
* The spectrum-like distribution of data over dosage PC1 still observable.
* PC2 changed after removing Library_33
```{r PCA_cDNA, echo=FALSE}
plotPCA(vsd, intgroup = c("dosage"))
plotPCA(vsd, intgroup = c("chip"))
plotPCA(vsd, intgroup = c("date"))
plotPCA(vsd, intgroup = c("folder"))
```

#### DESeq with Wald test
Some miRNAs have NA padj values (not shown). NA padj values are generated per mechanism "If a row is filtered by automatic independent filtering, for having a low mean normalized count, then only the adjusted p value will be set to NA."
```{r DESeq_w_cDNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_wald <- DESeq(dds)
resultsNames(dds_wald)
```
* dosage 50mg vs 0mg results (only print the top 10 cDNAs ordered by smallest p-value first, same for below)
```{r DESeq_wr1_cDNA, echo=FALSE}
res_wald_50 <- results(dds_wald, name="dosage_50_vs_0")
res_wald_50 <- res_wald_50[order(res_wald_50$pvalue),]
res_wald_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_wr2_cDNA, echo=FALSE}
res_wald_60 <- results(dds_wald, name="dosage_60_vs_0")
res_wald_60 <- res_wald_60[order(res_wald_60$pvalue),]
res_wald_60[1:10,]
```
* dosage 75mg vs 0mg results
```{r DESeq_wr3_cDNA, echo=FALSE}
res_wald_75 <- results(dds_wald, name="dosage_75_vs_0")
res_wald_75 <- res_wald_75[order(res_wald_75$pvalue),]
res_wald_75[1:10,]
```

#### Volcano plots for DESeq with Wald test
```{r volcano_wald_cDNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_wald_50,
    lab = rownames(res_wald_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_60,
    lab = rownames(res_wald_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_75,
    lab = rownames(res_wald_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```


#### DESeq with likelihood ratio test
Since we are running the log likelihood test, the p-values should be the same for all dosage groups. Log2FoldChange will be, however, different for each dosage group.
```{r DESeq_l_cDNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_lrt <- DESeq(dds, test="LRT", reduced=~1)
resultsNames(dds_lrt)
```
* dosage 50mg vs 0mg results (only print the top 10 cDNAs ordered by smallest p-value first, same for below)
```{r DESeq_lr1_cDNA, echo=FALSE}
res_lrt_50 <- results(dds_lrt, name="dosage_50_vs_0")
res_lrt_50 <- res_lrt_50[order(res_lrt_50$pvalue),]
res_lrt_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_lr2_cDNA, echo=FALSE}
res_lrt_60 <- results(dds_lrt, name="dosage_60_vs_0")
res_lrt_60 <- res_lrt_60[order(res_lrt_60$pvalue),]
res_lrt_60[1:10,]
```
* dosage 75mg vs 0mg results 
```{r DESeq_lr3_cDNA, echo=FALSE}
res_lrt_75 <- results(dds_lrt, name="dosage_75_vs_0")
res_lrt_75 <- res_lrt_75[order(res_lrt_75$pvalue),]
res_lrt_75[1:10,]
```


#### Volcano plots for DESeq with likelihood ratio test
```{r volcano_lrt_cDNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_lrt_50,
    lab = rownames(res_lrt_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_60,
    lab = rownames(res_lrt_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_75,
    lab = rownames(res_lrt_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```


#### heatmap of counts
* columns clustered 
* show cDNAs of top 20 mean counts
```{r DESeq_hm_cDNA, echo=FALSE}
select_wald <- order(rowMeans(counts(dds_wald, normalized=TRUE)),
                decreasing=TRUE)[1:20] # top 20
df_wald <- as.data.frame(colData(dds_wald)[,"dosage"])
colnames(df_wald) <- "dosage"
rownames(df_wald) <- colnames(dds_wald)
pheatmap(assay(vsd)[select_wald,], cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df_wald)
```





***
***






### ncRNA counts parsed from sRNAbench results {.tabset}
Repeat the above analysis on ncRNA counts parsed from sRNAbench results
```{r load_ncRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
readCounts <- read.csv(path_ncRNA)
genes <- readCounts[, 1]
readCounts <- data.matrix(readCounts[, -1])
libraries <- colnames(readCounts)

readCounts <- matrix(as.numeric(readCounts), ncol=length(libraries))
rownames(readCounts) <- genes
colnames(readCounts) <- libraries
readCounts <- readCounts[, order(colnames(readCounts))]
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_24")] # Remove library 24
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_33")] # Remove library 33

rownames(colData) <- colnames(readCounts)

dds <- DESeqDataSetFromMatrix(countData=readCounts, colData=colData, design=~dosage)

# pre-filtering
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]

vsd <- varianceStabilizingTransformation(dds, blind = FALSE)
```

#### sample-wise heatmap
* rows and columns clustered
* label = library+dosages

```{r sample-wise_heatmap_ncRNA, echo=FALSE}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste("Library",vsd$sample,"  -  Dosage",vsd$dosage)
colnames(sampleDistMatrix) <- NULL #vsd$dosage
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors, cluster_rows = T, cluster_cols = T)
```

#### PCA plots
* From top to bottom: (1) groups = dosage; (2) groups = chip; (3) groups = date; (4) groups = folder
* The results seem to suggest the batch effect of chip (see chip 6) / date (see 2018-04-18).
* The spectrum-like distribution of data over dosage PC1 still observable.
* Library_06 as an outlier?
```{r PCA_ncRNA, echo=FALSE}
plotPCA(vsd, intgroup = c("dosage"))
plotPCA(vsd, intgroup = c("chip"))
plotPCA(vsd, intgroup = c("date"))
plotPCA(vsd, intgroup = c("folder"))
```

#### DESeq with Wald test
Some miRNAs have NA padj values (not shown). NA padj values are generated per mechanism "If a row is filtered by automatic independent filtering, for having a low mean normalized count, then only the adjusted p value will be set to NA."
```{r DESeq_w_ncRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_wald <- DESeq(dds)
resultsNames(dds_wald)
```
* dosage 50mg vs 0mg results (only print the top 10 ncRNAs ordered by smallest p-value first, same for below)
```{r DESeq_wr1_ncRNA, echo=FALSE}
res_wald_50 <- results(dds_wald, name="dosage_50_vs_0")
res_wald_50 <- res_wald_50[order(res_wald_50$pvalue),]
res_wald_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_wr2_ncRNA, echo=FALSE}
res_wald_60 <- results(dds_wald, name="dosage_60_vs_0")
res_wald_60 <- res_wald_60[order(res_wald_60$pvalue),]
res_wald_60[1:10,]
```
* dosage 75mg vs 0mg results
```{r DESeq_wr3_ncRNA, echo=FALSE}
res_wald_75 <- results(dds_wald, name="dosage_75_vs_0")
res_wald_75 <- res_wald_75[order(res_wald_75$pvalue),]
res_wald_75[1:10,]
```

#### Volcano plots for DESeq with Wald test
```{r volcano_wald_ncRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_wald_50,
    lab = rownames(res_wald_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_60,
    lab = rownames(res_wald_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_75,
    lab = rownames(res_wald_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```


#### DESeq with likelihood ratio test
Since we are running the log likelihood test, the p-values should be the same for all dosage groups. Log2FoldChange will be, however, different for each dosage group.
```{r DESeq_l_ncRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_lrt <- DESeq(dds, test="LRT", reduced=~1)
resultsNames(dds_lrt)
```
* dosage 50mg vs 0mg results (only print the top 10 ncRNAs ordered by smallest p-value first, same for below)
```{r DESeq_lr1_ncRNA, echo=FALSE}
res_lrt_50 <- results(dds_lrt, name="dosage_50_vs_0")
res_lrt_50 <- res_lrt_50[order(res_lrt_50$pvalue),]
res_lrt_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_lr2_ncRNA, echo=FALSE}
res_lrt_60 <- results(dds_lrt, name="dosage_60_vs_0")
res_lrt_60 <- res_lrt_60[order(res_lrt_60$pvalue),]
res_lrt_60[1:10,]
```
* dosage 75mg vs 0mg results 
```{r DESeq_lr3_ncRNA, echo=FALSE}
res_lrt_75 <- results(dds_lrt, name="dosage_75_vs_0")
res_lrt_75 <- res_lrt_75[order(res_lrt_75$pvalue),]
res_lrt_75[1:10,]
```


#### Volcano plots for DESeq with likelihood ratio test
```{r volcano_lrt_ncRNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_lrt_50,
    lab = rownames(res_lrt_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_60,
    lab = rownames(res_lrt_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_75,
    lab = rownames(res_lrt_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```


#### heatmap of counts
* columns clustered 
* show ncRNAs of top 20 mean counts
```{r DESeq_hm_ncRNA, echo=FALSE}
select_wald <- order(rowMeans(counts(dds_wald, normalized=TRUE)),
                decreasing=TRUE)[1:20] # top 20
df_wald <- as.data.frame(colData(dds_wald)[,"dosage"])
colnames(df_wald) <- "dosage"
rownames(df_wald) <- colnames(dds_wald)
pheatmap(assay(vsd)[select_wald,], cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df_wald) # looks strange
```


***
***






### all RNA counts by another dataset parsed from sRNAbench results {.tabset}
Repeat the above analysis on a collection of different sRNA counts generated using another dataset. piRNA counts can be further extracted from the dataset.
```{r load_RNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
readCounts <- read.csv(path_RNAcentral)
genes <- readCounts[, 1]
readCounts <- data.matrix(readCounts[, -1])
libraries <- colnames(readCounts)

readCounts <- matrix(as.numeric(readCounts), ncol=length(libraries))
rownames(readCounts) <- genes
colnames(readCounts) <- libraries
readCounts <- readCounts[, order(colnames(readCounts))]
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_24")] # Remove library 24
readCounts = readCounts[, !colnames(readCounts) %in% c("Library_33")] # Remove library 33

rownames(colData) <- colnames(readCounts)

dds <- DESeqDataSetFromMatrix(countData=readCounts, colData=colData, design=~dosage)

# pre-filtering
keep <- rowSums(counts(dds)) >= 10
dds <- dds[keep,]

vsd <- varianceStabilizingTransformation(dds, blind = FALSE)
```


#### sample-wise heatmap
* rows and columns clustered
* label = library+dosages

```{r sample-wise_heatmap_RNA, echo=FALSE}
sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste("Library",vsd$sample,"  -  Dosage",vsd$dosage)
colnames(sampleDistMatrix) <- NULL #vsd$dosage
colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows = sampleDists,
         clustering_distance_cols = sampleDists,
         col = colors, cluster_rows = T, cluster_cols = T)
```

#### PCA plots
* From top to bottom: (1) groups = dosage; (2) groups = chip; (3) groups = date; (4) groups = folder
* The results seem to suggest the batch effect of chip (see chip 6) / date (see 2018-04-18).
* The spectrum-like distribution of data over dosage PC1 still observable.
* very much like the plot for cDNA, but PC2 did not change
```{r PCA_RNA, echo=FALSE}
plotPCA(vsd, intgroup = c("dosage"))
plotPCA(vsd, intgroup = c("chip"))
plotPCA(vsd, intgroup = c("date"))
plotPCA(vsd, intgroup = c("folder"))
```

#### DESeq with Wald test
Some miRNAs have NA padj values (not shown). NA padj values are generated per mechanism "If a row is filtered by automatic independent filtering, for having a low mean normalized count, then only the adjusted p value will be set to NA."
```{r DESeq_w_RNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_wald <- DESeq(dds)
resultsNames(dds_wald)
```
* dosage 50mg vs 0mg results (only print the top 10 RNAs ordered by smallest p-value first, same for below)
```{r DESeq_wr1_RNA, echo=FALSE}
res_wald_50 <- results(dds_wald, name="dosage_50_vs_0")
res_wald_50 <- res_wald_50[order(res_wald_50$pvalue),]
res_wald_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_wr2_RNA, echo=FALSE}
res_wald_60 <- results(dds_wald, name="dosage_60_vs_0")
res_wald_60 <- res_wald_60[order(res_wald_60$pvalue),]
res_wald_60[1:10,]
```
* dosage 75mg vs 0mg results
```{r DESeq_wr3_RNA, echo=FALSE}
res_wald_75 <- results(dds_wald, name="dosage_75_vs_0")
res_wald_75 <- res_wald_75[order(res_wald_75$pvalue),]
res_wald_75[1:10,]
```

#### Volcano plots for DESeq with Wald test
```{r volcano_wald_RNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_wald_50,
    lab = rownames(res_wald_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_60,
    lab = rownames(res_wald_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_wald_75,
    lab = rownames(res_wald_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```


#### DESeq with likelihood ratio test
Since we are running the log likelihood test, the p-values should be the same for all dosage groups. Log2FoldChange will be, however, different for each dosage group.
```{r DESeq_l_RNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
dds_lrt <- DESeq(dds, test="LRT", reduced=~1)
resultsNames(dds_lrt)
```
* dosage 50mg vs 0mg results (only print the top 10 RNAs ordered by smallest p-value first, same for below)
```{r DESeq_lr1_RNA, echo=FALSE}
res_lrt_50 <- results(dds_lrt, name="dosage_50_vs_0")
res_lrt_50 <- res_lrt_50[order(res_lrt_50$pvalue),]
res_lrt_50[1:10,]
```
* dosage 60mg vs 0mg results 
```{r DESeq_lr2_RNA, echo=FALSE}
res_lrt_60 <- results(dds_lrt, name="dosage_60_vs_0")
res_lrt_60 <- res_lrt_60[order(res_lrt_60$pvalue),]
res_lrt_60[1:10,]
```
* dosage 75mg vs 0mg results 
```{r DESeq_lr3_RNA, echo=FALSE}
res_lrt_75 <- results(dds_lrt, name="dosage_75_vs_0")
res_lrt_75 <- res_lrt_75[order(res_lrt_75$pvalue),]
res_lrt_75[1:10,]
```


#### Volcano plots for DESeq with likelihood ratio test
```{r volcano_lrt_RNA, echo=FALSE, warning=FALSE, results='hide', message=FALSE}
EnhancedVolcano(res_lrt_50,
    lab = rownames(res_lrt_50),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_60,
    lab = rownames(res_lrt_60),
    title = '60mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
EnhancedVolcano(res_lrt_75,
    lab = rownames(res_lrt_75),
    title = '50mg vs 0mg',
    x = 'log2FoldChange',
    y = 'pvalue')
```


#### heatmap of counts
* columns clustered 
* show RNAs of top 20 mean counts
```{r DESeq_hm_RNA, echo=FALSE}
select_wald <- order(rowMeans(counts(dds_wald, normalized=TRUE)),
                decreasing=TRUE)[1:20] # top 20
df_wald <- as.data.frame(colData(dds_wald)[,"dosage"])
colnames(df_wald) <- "dosage"
rownames(df_wald) <- colnames(dds_wald)
pheatmap(assay(vsd)[select_wald,], cluster_rows=FALSE, show_rownames=FALSE,
         cluster_cols=TRUE, annotation_col=df_wald) # looks strange
```